tools/link.py

```diff
@@ -620,6 +622,11 @@ def phase_linker_setup(options, state, newargs):
   if settings.OPT_LEVEL >= 1:
     default_setting('ASSERTIONS', 0)
 
+  if settings.GZIP_EMBEDDINGS:
+    settings.SINGLE_FILE = 1
```

Comment:
I wonder if there other cases where we might want to use/expose base64 embeddings other than `-sSINGLE_FILE`?  I'm not sure if there are today.   But it might be safer to require the explicit `-sSINGLE_FILE` here especially since its just major option to opt into.

How about instead we do:

```
if settings.GZIP_EMBEDDINGS and not settings.SINGLE_FILE:
   exit_with_error(`GZIP_EMBEDDINGS is not currently meaningfull except in SINGLE_FILE mode`)
```


The good thing about doing it this way is that we an always decide to remove this error and make `SINGLE_FILE` implicit later, but its hard to go the other way without breaking folks.

Reply:
Yeah, ur right, explicit, no sirprize.